openedx_external_enrollments/tests/external_enrollments/tests_salesforce_external_enrollment.py

```diff
@@ -222,97 +220,216 @@ def test_get_program_of_interest_data(self, get_course_mock, get_user_mock):
+        expected_data['CourseName'] = 'test-program'
+
+        self.assertEqual([expected_data], self.base._get_courses_data({}, lines))  # pylint: disable=protected-access
+
+        get_course_mock.side_effect = Exception('test-exception')
+
+        self.assertEqual([], self.base._get_courses_data({}, lines))  # pylint: disable=protected-access
+
+    @patch.object(SalesforceEnrollment, '_get_program_of_interest_data')
+    @patch.object(SalesforceEnrollment, '_get_program_course_runs')
+    @patch.object(SalesforceEnrollment, '_get_course_start_date')
+    @patch.object(SalesforceEnrollment, '_get_course_key')
+    @patch.object(SalesforceEnrollment, '_get_course')
+    def test_get_courses_data_case_2(
+        self, get_course_mock, get_course_key_mock, get_date_mock,
+        get_runs_mock, get_poi_mock
+    ):
+        """Testing _get_courses_data method for case 2 (1 program)."""
+        course_id = 'test-course-id'
+        lines = [
+            {
+                'user_email': 'test-email',
+                'course_id': course_id,
+            },
+        ]
+        data = {
+            'program': {
+                'courses': [
+                    {
+                        'course_runs': [
+                            {'key': course_id},
+                        ],
+                    },
+                ],
+            },
+        }
+        now = datetime.now()
+        now_date_format = now.strftime('%Y-%m-%d')
         course_mock = Mock()
-        course_mock.other_course_settings = {
-            'salesforce_data': expected_data,
+        course_mock.end = now
+        course_mock.display_name = 'test-course'
+        course_mock.other_course_settings = {}
+        get_poi_mock.return_value = {
+            'Institution_Hidden': 'HI_from_program',
+            'Program_of_Interest': 'POI_from_program',
         }
+        get_runs_mock.return_value = [course_id]
         get_course_mock.return_value = course_mock
+        get_course_key_mock.return_value = CourseKey.from_string('course-v1:PX+test-course-run-id+2015')
+        get_date_mock.return_value = now_date_format
+        expected_data = {
+            'CourseName': course_mock.display_name,
+            'CourseID': 'PX+test-course-run-id',
+            'CourseRunID': course_id,
+            'CourseStartDate': now_date_format,
+            'CourseEndDate': now_date_format,
+            'CourseDuration': '0',
+            'Institution_Hidden': 'HI_from_program',
+            'Program_of_Interest': 'POI_from_program',
+        }
 
-        program_data = self.base._get_program_of_interest_data(data, lines)  # pylint: disable=protected-access
-
-        self.assertEqual(program_data['Institution_Hidden'], expected_data['Institution_Hidden'])
-        self.assertEqual(program_data['Program_of_Interest'], expected_data['Program_of_Interest'])
+        self.assertEqual([expected_data], self.base._get_courses_data(data, lines))  # pylint: disable=protected-access
```

Comment:
@Jacatove, let's separate the tests to [Triple A pattern](https://3.basecamp.com/3966315/buckets/12732851/documents/2737536654) for all cases, please.